RHDEVDOCS-3956 - document per label scrape limit settings for UWM (#48050)
openshift/openshift-docs: modules/monitoring-limiting-scrape-samples-in-user-defined-projects.adoc

//
// * monitoring/configuring-the-monitoring-stack.adoc

:_content-type: CONCEPT
[id="controlling-the-impact-of-unbound-attributes-in-user-defined-projects_{context}"]
= Controlling the impact of unbound metrics attributes in user-defined projects

Developers can create labels to define attributes for metrics in the form of key-value pairs. The number of potential key-value pairs corresponds to the number of possible values for an attribute. An attribute that has an unlimited number of potential values is called an unbound attribute. For example, a `customer_id` attribute is unbound because it has an infinite number of possible values.

> Every assigned key-value pair has a unique time series. The use of many unbound attributes in labels can result in an exponential increase in the number of time series created. This can impact Prometheus performance and can consume a high amount of disk space.

Reviewer: ```suggestion
Every assigned key-value pair has a unique time series. Using many unbound attributes in labels can create exponentially more time series, which impact Prometheus performance and available disk space.
```

Author: Edited per you suggestions

Cluster administrators can use the following measures to control the impact of unbound metrics attributes in user-defined projects:

* Limit the number of samples that can be accepted per target scrape in user-defined projects
* Limit the number of scraped labels, the length of label names, and the length of label values.
* Create alerts that fire when a scrape sample threshold is reached or when the target cannot be scraped

[NOTE]
====

> Limiting scrape samples and label names can help prevent issues caused by adding many unbound attributes to labels. Developers can also prevent the underlying cause by limiting the number of unbound attributes that they define for metrics. Using attributes that are bound to a limited set of possible values reduces the number of potential key-value pair combinations.

Reviewer: ```suggestion
To prevent issues caused by adding many unbound attributes, limit the number of scrape samples, label names, and unbound attributes you define for metrics. Also reduce the number of potential key-value pair combinations by using attributes that are bound to a limited set of possible values.
```
How about rewriting this to be more user-action focused?

Author: > To prevent issues caused by adding many unbound attributes, limit the number of scrape samples, label names, and unbound attributes you define for metrics. Also reduce the number of potential key-value pair combinations by using attributes that are bound to a limited set of possible values.

Edited per your suggested wording.
